### DISCUSSION TITLE
How to change focus color?

### DISCUSSION THREAD

**Discussion by cktang88** [chakra-ui/chakra-ui#2459] [Q&A] [answered]
By default every component when clicked gets outlined in the light blue, could I change that color to be another color, ideally in the theme file?

**Comment by with-heart** [comment, marked as answer]
Hi @cktang88, you can modify `theme.shadows.outline`. See previous discussion in #1921 around the outlines.

**Reply by cktang88** [reply]
thanks!

**Comment by cevr** [comment]
Does this still work? Seems like setting theme.shadows.outline completely removes all outlines in version 1.7.2

**Reply by blair-grylls** [reply]
Hey I had a similar question and found an answer from the discord:

https://codesandbox.io/s/chakra-change-focus-ring-color-c2yis?file=/src/theme.ts

theme.shadows.outline is actually acting as a box-shadow so you need to change what argument you are giving it. It also apparently doesn't work for all components by default so there is some variants covering the edge cases in the theme.ts file

**Reply by cevr** [reply]
Thank you!

**Comment by hevans90** [comment]
Hey, using something like this works for everything EXCEPT Chakra `Input` components, likely because they have `defaultProps` set:

```
extendTheme({
  shadows: { outline: '0 0 0 3px var(--chakra-colors-green-500)' },
});
```

To fix this, in your `theme` file import all the input types you want to restyle and:

```
Input.defaultProps = { ...Input.defaultProps, focusBorderColor: 'green.500' };
NumberInput.defaultProps = { ...NumberInput.defaultProps, focusBorderColor: 'green.500' };
```

**Comment by hillmark** [comment]
I used the above approach suggested by @hevans90, but used the `components` prop to override `defaultProps` to avoid the need for imports:
```
extendTheme({
  shadows: { outline: '0 0 0 2px var(--chakra-colors-cyan-500)' },
  components: {
    NumberInput: {
      defaultProps: {
        focusBorderColor: 'cyan.500',
      }
    }
  }
})
```
